src/binary/mod.rs

```diff
@@ -246,6 +247,13 @@ where
         None
     };
 
+    let kernel_info = KernelInfo {
+        kernel_base: kernel_bytes.as_ptr() as u64,
+        kernel_size: kernel_bytes.len() as u64,
```

Comment:
This is the size of the kernel ELF file, not the size that the kernel occupies in memory. Also, as noted in the previous comment, the kernel segments can be spread across the whole virtual address space, so I'm not sure if we can even choose a proper `kernel_size` for all kernels.